docs/articles_en/openvino_workflow/gen_ai.rst

```diff
@@ -135,9 +135,17 @@ The optimized model can be saved as usual with a call to ``save_pretrained()``.
 
 .. note::
 
-   OpenVINO also supports 4-bit models from Hugging Face `Transformers <https://github.com/huggingface/transformers>`__ library optimized 
+   OpenVINO also supports 4-bit models from Hugging Face `Transformers <https://github.com/huggingface/transformers>`__ library optimized
    with `GPTQ <https://github.com/PanQiWei/AutoGPTQ>`__. In this case, there is no need for an additional model optimization step because model conversion will automatically preserve the INT4 optimization results, allowing model inference to benefit from it.
 
+Another optimization that is applied by default when using ``OVModelForCausalLM`` class is transformation of the model to a stateful form.
+This transformation further improves inference performance and decreases amount of allocated runtime memory in long running text generation scenarious.
+It is achieved by hiding inputs and outputs of the model that represent past KV-cache tensors, and handling them inside the model in a more efficient way.
+This feature is activated automatically for a wide range of supported text generation models, keeping not supported models in a regular, stateless form.
+
+Model usage are identical for stateful and stateless models as long as Optimum-Intel API is used because KV-cache handling is an internal detail of how Optimum-Intel is implmented under the hood.
```

Comment:
```suggestion
Model usage are identical for stateful and stateless models as long as Optimum-Intel API is used because KV-cache handling is an internal detail of the text-generation API of Transformers library.
```